esp-hal-procmacros/src/lib.rs

```diff
@@ -381,3 +396,116 @@ pub fn main(args: TokenStream, item: TokenStream) -> TokenStream {
 
     run(&args.meta, f, main()).unwrap_or_else(|x| x).into()
 }
+
+/// Automatically implement the [Builder Lite] pattern for a struct.
+///
+/// This will create an `impl` which contains methods for each field of a
+/// struct, allowing users to easily set the values. The generated methods will
+/// be the field name prefixed with `with_`, and calls to these methods can be
+/// chained as needed.
+///
+/// ## Example
+///
+/// ```rust, no_run
+/// #[derive(Default)]
+/// enum MyEnum {
+///     #[default]
+///     A,
+///     B,
+/// }
+///
+/// #[derive(Default, BuilderLite)]
+/// #[non_exhaustive]
+/// struct MyStruct {
+///     enum_field: MyEnum,
+///     bool_field: bool,
+///     option_field: Option<i32>,
+/// }
+///
+/// MyStruct::default()
+///     .with_enum_field(MyEnum::B)
+///     .with_bool_field(true)
+///     .with_option_field(-5);
```

Comment:
I'm not sure this is an entirely good idea. What if the default is `Some(value)` and the user wants `None`? I think we shouldn't preprocess options, just handle them the same way the other types are.

Reply:
No strong feelings, based on my experience this is usually the approach take by builder patterns, so this is what I did. No issues changing it if there is consensus to do so.

Reply:
Do we have any `Default` impls which actually set a value for an `Option`?

Reply:
I2c Config sets a default timeout, yes.

Reply:
if we really want to, we could generate a `with_timeout_none()` function instead 🤷‍♂️

Reply:
Any solution that lets me reset a value back to None is fine. Users may call these functions to change an existing configuration as well even if that is a bit awkward.

Reply:
My position in the meeting was to follow the builder lite pattern and take the value directly for `Option`s, which I still stand by. I think it might be worth exploring `with_optional_none` that @bjoernQ suggested, because it would be nice to keep the method chaining in the case the user wants to set a timeout to `None`.